src/interactable_ice_paradise_slowing_snow.c

```diff
@@ -0,0 +1,115 @@
+#include "global.h"
+
+#include "task.h"
+#include "game.h"
+
+#include "interactable.h"
+#include "sprite.h"
+
+#include "constants/move_states.h"
+
+typedef struct {
+    /* 0x00 */ u8 x;
+    /* 0x01 */ u8 y;
+    /* 0x02 */ u8 index;
+
+    /* 0x03 */ s8 offsetX;
+    /* 0x04 */ s8 offsetY;
+    /* 0x05 */ u8 width;
+    /* 0x06 */ u8 height;
+} Interactable_SlowingSnow PACKED;
+
+typedef struct {
+    /* 0x00 */ s16 left;
+    /* 0x02 */ s16 top;
+    /* 0x04 */ s16 right;
+    /* 0x06 */ s16 bottom;
+    /* 0x08 */ s32 posX;
+    /* 0x0C */ s32 posY;
+    /* 0x10 */ Interactable_SlowingSnow *ia;
+    /* 0x14 */ u8 spriteX;
+    /* 0x15 */ u8 spriteY;
+} Sprite_SlowingSnow; // size = 0x18
+
+static void Task_Interactable_IceParadise_SlowingSnow(void);
+static void TaskDestructor_Interactable_IceParadise_SlowingSnow(struct Task *t);
+
+extern bool32 sub_8077F18(Sprite_SlowingSnow *);
+
+void initSprite_Interactable_IceParadise_SlowingSnow(Interactable *in_ia,
+                                                     u16 spriteRegionX,
+                                                     u16 spriteRegionY, u8 spriteY)
+{
+    struct Task *t = TaskCreate(Task_Interactable_IceParadise_SlowingSnow,
+                                sizeof(Sprite_SlowingSnow), 0x2010, 0,
+                                TaskDestructor_Interactable_IceParadise_SlowingSnow);
+
+    Sprite_SlowingSnow *snow = TaskGetStructPtr(t);
+    Interactable_SlowingSnow *ia = (Interactable_SlowingSnow *)in_ia;
+    snow->left = ia->offsetX * 8;
+    snow->top = ia->offsetY * 8;
+    snow->right = snow->left + ia->width * 8;
+    snow->bottom = snow->top + ia->height * 8;
+    snow->ia = ia;
+    snow->spriteX = ia->x;
+    snow->spriteY = spriteY;
+
+    snow->posX = SpriteGetScreenPos(ia->x, spriteRegionX);
+    snow->posY = SpriteGetScreenPos(ia->y, spriteRegionY);
+    SET_SPRITE_INITIALIZED(ia);
+}
+
+bool32 PlayerTouchesSnow(Sprite_SlowingSnow *snow)
```

Comment:
Minor point, prefer `PlayerTouchingSnow`, `Touches` doesn't provide a time context, and thus questions could be asked "Does this function return true if the player ever touches snow?"

Reply:
I mean... it's not like there's tons of abstractions, the procudure is literally right next to it. :D
But will try to keep it in mind in the future

Reply:
But I def should make it static